server/routes/helpers.js

```diff
@@ -59,7 +62,49 @@ function wrapper(handler) {
   };
 }
 
+async function generateToTPSecret(email, transporter) {
+  const secret = new OTPAuth.Secret();
+  // new TOTP object using the secret key
+  const totp = new OTPAuth.TOTP({
+    secret: secret.base32,
+    issuer: 'Routed',
+    period: -1,
+    digits: 6,
+  });
+  // save the secret key to the session
+  // generate secret token
+  const token = totp.generate();
+  // Save token and expiration timestamp in DB (Expires in 15 minutes from inital Log In)
+  const user = await models.User.findOne({
+    where: { email: email },
+  });
+  // save totp secret and expiration timestamp in DB
+  await user.update({ twoFactorData: { totptimestamp: Date.now() + 900000, totptoken: token } });
+  await user.save();
+  // send email with token
+  sendMail(
+    transporter,
+    email,
+    'Your Authentication Code from Routed',
+    `This is your Authentication Code: ${token} . It will expire in 15 minutes.`
+  );
+}
+async function verifyTwoFactor(req) {
+  const token = req.body.code;
+  const email = req.user.dataValues.email;
+  const user = await models.User.findOne({
+    where: { email: email },
+  });
+
+  const totptoken = user.dataValues.twoFactorData.totptoken;
+  const totptimestamp = user.dataValues.twoFactorData.totptimestamp;
+  const verified = token === totptoken && Date.now() < totptimestamp;
+
+  return verified;
+}
```

Comment:
This helpers file contains functions that are used by many different routes... this two-factor code seems very user/login specific and maybe not the best place for it.

It feels like these can be instance methods on the User model class. The User object instance for the user is already loaded in `req.user`, so you wouldn't need to do another db query/fetch.